hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/cache/CacheManager.java

```diff
@@ -85,141 +85,150 @@ public void init() {
-                    publishCache.entrySet().stream().forEach(e -> log.tracef("Published: %s", e.getValue()));
-                }
-
-                // This will include group trigger conditions, which is OK because for data-driven group triggers the
-                // dataIds will likely be the dataIds from the group level, made distinct by the source.
-                Collection<Condition> conditions = definitions.getAllConditions();
-                final Set<CacheKey> activeKeys = new HashSet<>();
-                for (Condition c : conditions) {
-                    CacheKey cacheKey = new CacheKey(c.getTenantId(), c.getDataId());
-                    if (!activeKeys.contains(cacheKey)) {
-                        activeKeys.add(cacheKey);
-                        if (!currentlyPublished.contains(cacheKey)) {
-                            publish(cacheKey);
-                        }
-                    }
-                    if (c instanceof CompareCondition) {
-                        String data2Id = ((CompareCondition) c).getData2Id();
-                        CacheKey cacheKey2 = new CacheKey(c.getTenantId(), data2Id);
-                        if (!activeKeys.contains(cacheKey2)) {
-                            activeKeys.add(cacheKey2);
-                            if (!currentlyPublished.contains(cacheKey2)) {
-                                publish(cacheKey2);
-                            }
-                        }
-                    }
-
-                }
-
-                final Set<CacheKey> doomedKeys = new HashSet<>();
-                if (!currentlyPublished.isEmpty()) {
-                    currentlyPublished.stream()
-                            .filter(k -> !activeKeys.contains(k))
-                            .forEach(k -> doomedKeys.add(k));
-                }
-                unpublish(doomedKeys);
-
-                log.debugf("Published after update=%s", publishCache.size());
-                if (log.isTraceEnabled()) {
-                    publishCache.entrySet().stream().forEach(e -> log.tracef("Published: %s", e.getValue()));
+            log.debug("Initial Cache update in progress..");
+
+            // This will include group trigger conditions, which is OK because for data-driven group triggers the
+            // dataIds will likely be the dataIds from the group level, made distinct by the source.
+            Collection<Condition> conditions = definitions.getAllConditions();
+            for (Condition c : conditions) {
+                String triggerId = c.getTriggerId();
+                CacheKey cacheKey = new CacheKey(c.getTenantId(), c.getDataId());
+                initialPublish(cacheKey, triggerId);
+                if (c instanceof CompareCondition) {
+                    String data2Id = ((CompareCondition) c).getData2Id();
+                    CacheKey cacheKey2 = new CacheKey(c.getTenantId(), data2Id);
+                    initialPublish(cacheKey2, triggerId);
                 }
             }
```

Comment:
I guess my big question here is why we don't have to populate dataIdsCache in this initialCacheUpdate.  Don't we need both caches fully defined for the mechanism to work?